src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnectionString.cs

```diff
@@ -22,14 +22,6 @@ internal OdbcConnectionString(string connectionString, bool validate) : base(con
                 int position = 0;
                 _expandedConnectionString = ExpandDataDirectories(ref filename, ref position);
             }
-            if (validate || (null == _expandedConnectionString))
-            {
-                // do not check string length if it was expanded because the final result may be shorter than the original
-                if ((null != connectionString) && (ODBC32.MAX_CONNECTION_STRING_LENGTH < connectionString.Length))
-                { // MDAC 83536
```

Comment:
Do we even know what this was? :)

Reply:
We tried finding out - for months (I can send you some email threads if you're really interested). One theory @GrabYourPitchforks had (based on other modifications in the changeset) was that this was a localization thing, i.e. that the P/Invoke call failed for long connection strings with some cryptic/non-localized error message, and that this check was added to surface a better error message. But we didn't manage to find a definitive answer.